### PR TITLE
feat: add drizzle database connection as middleware

### PR DESCRIPTION
**Pull request by vincentdo1**

### CODE REVIEW

File: hono/src/service/items.ts [hunk 1]
```diff
@@ -90,22 +87,29 @@ app.put(
...
     );
   },
 );
 
 app.delete('/:id', async (c) => {
   const id: number = +c.req.param('id');
 
-  const db = drizzle(c.env.DB);
+  await c
+    .get('db')
```
[the remark this answers is not in the document]
Comment (vincentdo1) [reply hono/src/service/items.ts:105]: Yes, c.get('db') gets a database that has already been instantiated.

[the remark this answers is not in the document]
Comment (rak3rman) [reply hono/src/service/items.ts:105]: I also verified behavior, that's an interesting and kinda cool feature.

File: hono/src/service/items.ts [hunk 2]
```diff
@@ -21,14 +20,14 @@ const insertItemSchema = createInsertSchema(items, {
...
+
 app.get('/', async (c) => {
   const type: number = Number(c.req.query('type') || -1);
   const search: string = c.req.query('search') || '';
 
-  const db = drizzle(c.env.DB, { schema });
-
   return c.json(
-    await db.query.items.findMany({
+    await c.get('db').query.items.findMany({
```
[the remark this answers is not in the document]
Comment (vincentdo1) [reply hono/src/service/items.ts:30]: I have tried specifying the return type for `c.get` function as suggested in the link but I still got typing errors. The only solution I was able to come up with was specifying type `c` to be `Context` but it seems other typing errors occur that I was not able to find a clean solution for. Let me know what you think.

[the remark this answers is not in the document]
Comment (rak3rman) [reply hono/src/service/items.ts:30]: Let me take a stab at it.

File: hono/src/service/items.ts [hunk 3]
```diff
@@ -1,42 +1,34 @@
...
 
-app.get('/', async (c) => {
-  const type: number = Number(c.req.query('type') || -1);
-  const search: string = c.req.query('search') || '';
+app.get('/', injectDB, async (c) => {
+  const type: number = Number(c.req.query('type') || -1)
+  const search: string = c.req.query('search') || ''
 
-  const db = drizzle(c.env.DB, { schema });
+  const db = createDB(c)
```
Comment (rak3rman) [inline hono/src/service/items.ts:23]: This shouldn't be here :(

**Review by rak3rman — CHANGES_REQUESTED** [review]

File: hono/src/service/items.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (rak3rman) [reply hono/src/service/items.ts:24]: Marking as resolved, going to open another ticket for weak warnings that are appearing.

**Review by rak3rman — APPROVED** [review]

**Review by SengdowJones — APPROVED** [review]
LGTM!👍

**Pull request merged**